src/main/TreeQuery.cs

```diff
@@ -0,0 +1,200 @@
+                        };
+
+                        result2 = true;
+                    }
+                    catch (ArgumentException)
+                    {
+                        // Handle invalid enum values gracefully
+                        result = null;
+                        result2 = false;
+                    }
+                }
+            }
+
+            return result2;
+        }
+
+        private static Dictionary<string, List<string>> TreeQueryParseQueryString(string queryString)
+        {
+            var parameters = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
+            string[] pairs = queryString.Split('&');
+
+            foreach (string pair in pairs)
+            {
+                if (string.IsNullOrEmpty(pair)) continue;
+
+                int equalIndex = pair.IndexOf('=');
+                if (equalIndex > 0)
+                {
+                    string key = HttpUtility.UrlDecode(pair.Substring(0, equalIndex));
+                    string val = HttpUtility.UrlDecode(pair.Substring(equalIndex + 1));
+
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add(val);
+                }
+                else if (equalIndex == 0)
+                {
+                    // Handle case where key is empty but there's a value
+                    continue;
+                }
+                else
+                {
+                    // Handle case where there's no equals sign (key without value)
+                    string key = HttpUtility.UrlDecode(pair);
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add("");
+                }
+            }
+
+            return parameters;
+        }
+
+        private static List<string> TreeQueryConvertParametersToArray(Dictionary<string, List<string>> parameters)
+        {
+            var parameterArray = new List<string>();
+            foreach (var kvp in parameters)
+            {
+                foreach (var val in kvp.Value)
+                {
+                    parameterArray.Add($"{kvp.Key}={val}");
+                }
+            }
+            return parameterArray;
+        }
+
+        private static IEnumerable<Guid> TreeQueryParseGuidArray(IEnumerable<string> stringValues)
+        {
+            if (stringValues == null)
+                return null;
+
+            var guids = new List<Guid>();
+            foreach (string value in stringValues)
+            {
+                if (Guid.TryParse(value, out Guid guid))
+                {
+                    guids.Add(guid);
+                }
+            }
+            return guids.Any() ? guids : null;
+        }
+
+        private static string TreeQueryGetQuerySingleValue(string[] array, string key)
```

Comment:
Is this method necessary? Can't we just call something similar to the result of HttpUtility.ParseQueryString? eg. paramsCollection[key]

https://learn.microsoft.com/en-us/dotnet/api/system.web.httputility.parsequerystring?view=net-9.0